yarn-project/end-to-end/src/e2e_deploy_contract.test.ts

```diff
@@ -46,226 +43,190 @@ describe('e2e_deploy_contract', () => {
+        portalContract.toString(),
       );
+    }, 60_000);
+
+    // TODO(@spalladino): Review this test, it's showing an unexpected 'Bytecode not found' error in logs.
+    // It's possible it is failing for the wrong reason, and the getContractData checks are returning wrong data.
+    it('should not deploy a contract which failed the public part of the execution', async () => {
+      sequencer?.updateSequencerConfig({
+        minTxsPerBlock: 2,
+      });
 
-      await Promise.all([
-        goodDeploy.simulate({ skipPublicSimulation: true }),
-        badDeploy.simulate({ skipPublicSimulation: true }),
-      ]);
+      try {
+        // This test requires at least another good transaction to go through in the same block as the bad one.
+        // I deployed the same contract again but it could really be any valid transaction here.
+        const artifact = TokenContractArtifact;
+        const initArgs = ['TokenName', 'TKN', 18] as const;
+        const goodDeploy = new ContractDeployer(artifact, wallet).deploy(AztecAddress.random(), ...initArgs);
+        const badDeploy = new ContractDeployer(artifact, wallet).deploy(AztecAddress.ZERO, ...initArgs);
 
-      const [goodTx, badTx] = [
-        goodDeploy.send({ skipPublicSimulation: true }),
-        badDeploy.send({ skipPublicSimulation: true }),
-      ];
+        const firstOpts = { skipPublicSimulation: true };
+        const secondOpts = { skipPublicSimulation: true, skipClassRegistration: true };
 
-      const [goodTxPromiseResult, badTxReceiptResult] = await Promise.allSettled([goodTx.wait(), badTx.wait()]);
+        await Promise.all([goodDeploy.simulate(firstOpts), badDeploy.simulate(secondOpts)]);
+        const [goodTx, badTx] = [goodDeploy.send(firstOpts), badDeploy.send(secondOpts)];
+        const [goodTxPromiseResult, badTxReceiptResult] = await Promise.allSettled([goodTx.wait(), badTx.wait()]);
 
-      expect(goodTxPromiseResult.status).toBe('fulfilled');
-      expect(badTxReceiptResult.status).toBe('rejected');
+        expect(goodTxPromiseResult.status).toBe('fulfilled');
+        expect(badTxReceiptResult.status).toBe('rejected');
 
-      const [goodTxReceipt, badTxReceipt] = await Promise.all([goodTx.getReceipt(), badTx.getReceipt()]);
+        const [goodTxReceipt, badTxReceipt] = await Promise.all([goodTx.getReceipt(), badTx.getReceipt()]);
 
-      expect(goodTxReceipt.blockNumber).toEqual(expect.any(Number));
-      expect(badTxReceipt.blockNumber).toBeUndefined();
+        expect(goodTxReceipt.blockNumber).toEqual(expect.any(Number));
+        expect(badTxReceipt.blockNumber).toBeUndefined();
 
-      await expect(pxe.getExtendedContractData(goodDeploy.instance!.address)).resolves.toBeDefined();
-      await expect(pxe.getExtendedContractData(goodDeploy.instance!.address)).resolves.toBeDefined();
+        await expect(pxe.getExtendedContractData(goodDeploy.getInstance().address)).resolves.toBeDefined();
```

Comment:
Isn't this the same assertion twice?

Reply:
Good catch. Fixed and rebased on top of master.